### DISCUSSION TITLE
How to implement a context menu ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [floating-ui/floating-ui#2329] [React] [answered]
I can't find the task development information about the right-click context menu. I don't know how to implement it. Can I provide a demo?

**Comment by atomiks** [comment, marked as answer]
An example: https://codesandbox.io/s/trusting-rui-2duieo?file=/src/ContextMenu.tsx

**Reply by vaynevayne** [reply]
thanks evey much

**Reply by abelHodasz** [reply]
I'm having difficulty transforming the example, to only add the contextmenu and mouseup event listeners to a specific element, not the entire document. How would you go about it?

**Reply by Akiyamka** [reply]
https://gist.github.com/Akiyamka/f2b43cfe866e2f6f0b717409e975c61d

**Reply by IPv6** [reply]
> I'm having difficulty transforming the example, to only add the contextmenu and mouseup event listeners to a specific element, not the entire document. How would you go about it?

found this via google, hit the same problem... solution is to change "document.addEventListener ... document.removeEventListener" to use proper DOM element
